packages/cmd/src/services/cache/set.ts

```diff
@@ -59,28 +59,63 @@ export async function handleSetCache() {
   });
   debug('Cache upload url created', { result });
 
-  if (uploadPaths.length > 0) {
-    await handleArchiveUpload({
-      archive: await zipFilesToBuffer(uploadPaths),
-      cacheId: result.cacheId,
-      uploadUrl: result.uploadUrl,
-    });
+  const ciForMeta = getCIForMeta(ci);
+
+  const archive =
+    uploadPaths.length > 0 ? await zipFilesToBuffer(uploadPaths) : null;
+  const uploads = [];
 
-    info(uploadPaths.map((path) => `${dim('- uploading')} ${path}`).join('\n'));
+  if (archive) {
+    uploads.push(
+      handleArchiveUpload({
+        archive,
+        cacheId: result.cacheId,
+        uploadUrl: result.uploadUrl,
+      }).then(() => {
+        logUploadPaths(uploadPaths);
+        info('Cache archive uploaded', {
+          uploadUrl: result.uploadUrl,
+        });
+      })
+    );
   }
 
-  await handleMetaUpload({
-    meta: createMeta({
+  uploads.push(
+    handleMetaUpload({
+      meta: createMeta({
+        cacheId: result.cacheId,
+        config: config.values,
+        ci: ciForMeta,
+        orgId: result.orgId,
+        path: uploadPaths,
+      }),
       cacheId: result.cacheId,
-      config: config.values,
-      ci: getCIForMeta(ci),
-      orgId: result.orgId,
-      path: uploadPaths,
-    }),
-    cacheId: result.cacheId,
-    uploadUrl: result.metaUploadUrl,
-  });
+      uploadUrl: result.metaUploadUrl,
+    }).then(() =>
+      info('Meta uploaded', {
+        uploadUrl: result.metaUploadUrl,
+      })
+    ),
+    handleMetaUpload({
+      meta: createMeta({
+        cacheId: result.cacheId,
+        config: config.values,
+        ci: ciForMeta,
+        orgId: result.orgId,
+        path: uploadPaths,
+        archiveUrl: result.uploadUrl,
+        metaUrl: result.metaUploadUrl,
```

Comment:
This should be the keys for these, and not their urls (or at least  not their signed urls). The signed urls won't be good for long, so we still need the client to later take this info and fetch signed urls from the cache endpoint.